src/main/java/com/spotify/docker/client/ImageRef.java

```diff
@@ -44,18 +49,39 @@ public ImageRef(final String image) {
         this.tag = null;
       }
     }
+    final String[] parts = image.split("/", 2);
+    if (parts.length > 1 && isRegistry(parts[0])) {
+      this.registry = parts[0];
+    } else {
+      this.registry = DEFAULT_REGISTRY;
+    }
```

Comment:
this was added here to ImageRef to make the job of `ConfigFileRegistryAuthSupplier.authFor(imagename)`  easier